action/action-util/src/main/java/com/powsybl/action/util/GeneratorModificationTask.java

```diff
@@ -0,0 +1,162 @@
+import com.powsybl.contingency.tasks.ModificationTask;
+import com.powsybl.iidm.network.Generator;
+import com.powsybl.iidm.network.Network;
+
+import java.util.Objects;
+
+/**
+ * @author Olivier Perrin <olivier.perrin at rte-france.com>
+ */
+public class GeneratorModificationTask implements ModificationTask {
+
+    private final String generatorId;
+    private final Modifs modifs;
+
+    public GeneratorModificationTask(String generatorId, Modifs modifs) {
+        this.generatorId = Objects.requireNonNull(generatorId);
+        this.modifs = modifs;
+    }
+
+    @Override
+    public void modify(Network network, ComputationManager computationManager) {
+        Generator g = network.getGenerator(generatorId);
+        if (g == null) {
+            throw new PowsyblException("Generator '" + generatorId + "' not found");
+        }
+        if (modifs.getMinP() != null) {
+            g.setMinP(modifs.getMinP());
+        }
+        if (modifs.getMaxP() != null) {
+            g.setMaxP(modifs.getMaxP());
+        }
+
+        if (modifs.getVoltageRegulatorOn() != null) {
+            g.setVoltageRegulatorOn(modifs.getVoltageRegulatorOn());
+        }
+
+        boolean skipOtherConnectionChange = false;
+        if (modifs.getConnected() != null) {
+            changeConnectionState(g, modifs.getConnected());
+            skipOtherConnectionChange = true;
+        }
+
+        if (modifs.getTargetP() != null) {
+            setTargetpWithinBoundaries(g, modifs.getTargetP(), skipOtherConnectionChange);
+        } else if (modifs.getpDelta() != null) {
+            setTargetpWithinBoundaries(g, g.getTargetP() + modifs.getpDelta(), skipOtherConnectionChange);
+        }
+
+        if ((modifs.getTargetV() != null) && (g.isVoltageRegulatorOn())) {
+            // The target voltage can be defined only when the generator is in voltage regulation mode.
+            g.setTargetV(modifs.getTargetV());
+        }
+        if ((modifs.getTargetQ() != null) && (!g.isVoltageRegulatorOn())) {
+            // The target reactive can be defined only when the generator is NOT in voltage regulation mode.
+            g.setTargetQ(modifs.getTargetQ());
+        }
+    }
+
+    private void changeConnectionState(Generator g, boolean connect) {
+        if (connect) {
+            if (!g.getTerminal().isConnected()) {
+                GeneratorUtil.connectGenerator(g);
+            }
+        } else {
+            if (g.getTerminal().isConnected()) {
+                g.getTerminal().disconnect();
+            }
+        }
+    }
+
+    private void setTargetpWithinBoundaries(Generator g, double targetP, boolean skipConnect) {
+        if (!skipConnect) {
+            changeConnectionState(g, true);
+        }
+        g.setTargetP(Math.min(g.getMaxP(), Math.max(g.getMinP(), targetP)));
+    }
+
+    public static class Modifs {
+        private Double minP;
+        private Double maxP;
+        private Double targetP;
+        private Double pDelta;
```

Comment:
Open question : maybe `deltaP` would sound more consistent with other properties ? or `deltaTargetP` for a little more precision ?

Reply:
yes, deltaTargetP seems more consistent with all properties